src/utilities/scroll-utils.test.ts

```diff
@@ -36,6 +36,35 @@ describe("ScrollUtils", () => {
             // Assert
             expect(getElementByIdMock).toBeCalledTimes(50);
         });
+
+        test("when element is not found after 50 attempts, it logs a console warning for development environment", () => {
+            //Arrange
+            const elementId = faker.random.uuid();
+            process.env.NODE_ENV = "development";
+            const consoleWarnMock = jest.spyOn(console, "warn");
+
+            // Act
+            jest.useFakeTimers();
+            ScrollUtils.scrollToElementById(elementId);
+            jest.runAllTimers();
+
+            // Assert
+            expect(consoleWarnMock).toBeCalled();
+        });
+
+        test("when scrollOption has initial delay, it calls setTimeout with supplied delay", () => {
+            //Arrange
+            const options = { initialDelay: 200 };
+            const elementId = faker.random.uuid();
+            const setTimeoutMock = jest.spyOn(window, "setTimeout");
+
+            // Act
+            ScrollUtils.scrollToElementById(elementId, options);
+
+            // Assert
+            expect(setTimeoutMock).toBeCalled();
+            expect(setTimeoutMock.mock.calls[0][1]).toBe(options.initialDelay);
```

Comment:
I think you should consider using `expect(setTimeoutMock).toBeCalledWith (elementId, expect.objectContaining({initial delay: options.initialDelay}));`

Reply:
☝️ I agree, using the `toBeCalledWith` matcher would be more intuitive